tests/e2e/vsphere.go

```diff
@@ -665,6 +666,61 @@ func (vs *vSphere) getHostUUID(ctx context.Context, hostInfo string) string {
 	return ""
 }
 
+//getVsanClusterResource returns the vsan cluster's details
+func (vs *vSphere) getVsanClusterResource(ctx context.Context) *object.ClusterComputeResource {
+
+	finder := find.NewFinder(vs.Client.Client, false)
+	dc, err := finder.Datacenter(ctx, vs.Config.Global.Datacenters)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	finder.SetDatacenter(dc)
+
+	clusterComputeResource, err := finder.ClusterComputeResourceList(ctx, "*")
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	cluster := clusterComputeResource[0]
+
+	//TKG setup with NSX has edge-cluster enabled, this check is to skip that cluster
+	if strings.Contains(strings.ToLower(cluster.Name()), "edge") {
+		cluster = clusterComputeResource[1]
+	}
+	return cluster
+}
+
+//getAllHostsIP reads cluster, gets hosts in it and returns IP array
+func getAllHostsIP(ctx context.Context, vs *vSphere) []string {
+	var result []string
+	cluster := vs.getVsanClusterResource(ctx)
+	hosts, err := cluster.Hosts(ctx)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	for counter := range hosts {
+		framework.Logf("host name from hostUUID %v", hosts[counter].Name())
+		result = append(result, hosts[counter].Name())
+	}
+	return result
+}
+
+//getHostConnectionState reads cluster, gets hosts in it and returns power state of host
+func getHostConnectionState(ctx context.Context, vs *vSphere, addr string) (string, error) {
+	var state string
+	cluster := vs.getVsanClusterResource(ctx)
+	hosts, err := cluster.Hosts(ctx)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	for counter := range hosts {
+		var moHost mo.HostSystem
+		if hosts[counter].Name() == addr {
+			err = hosts[counter].Properties(ctx, hosts[counter].Reference(), []string{"summary"}, &moHost)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred(), "failed to get host system properties")
+			framework.Logf("Host owning IP: %v, and its power state is %s", hosts[counter].Name(),
```

Comment:
replace power with connection
`"Host owning IP: %v, and its power state is %s"` -> `"VC connection state for host %v, is %s"`

Reply:
Done